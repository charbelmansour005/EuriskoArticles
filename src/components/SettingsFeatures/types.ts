export interface FunctionProps {
  hideModal?: () => void
  testToast?: () => void
}
export interface SettingsFeaturesProps extends FunctionProps {
  chosenRippleColor?: string
  isVisible?: boolean
}
