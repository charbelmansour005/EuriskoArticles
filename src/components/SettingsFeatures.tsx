import {StyleSheet, Text, View, Modal} from 'react-native';
import {Portal, TouchableRipple} from 'react-native-paper';
import {themeColors} from '../helpers/themeColors';
import React from 'react';

interface FunctionProps {
  hideModal: () => void;
  testToast: () => void;
}
interface Props extends FunctionProps {
  chosenRippleColor?: string;
  isVisible?: boolean;
}

const SettingsFeatures = ({...props}: Props): JSX.Element => {
  return (
    <Portal>
      <Modal visible={props.isVisible} onDismiss={props.hideModal}>
        <View style={styles.featuresInfo}>
          <Text
            style={{
              color: 'gray',
              ...styles.featureText,
            }}>
            Hold an article to read more or visit website
          </Text>
          <Text
            style={{
              color: 'gray',
              ...styles.featureText,
            }}>
            Press Read more on an article to open more details
          </Text>
          <Text
            style={{
              color: props.chosenRippleColor,
              ...styles.featureText,
            }}>
            Dynamic colors
          </Text>
          <Text
            onPress={props.testToast}
            style={{
              color: 'gray',
              ...styles.featureText,
            }}>
            A Welcome Toast (Press to test)
          </Text>
        </View>
        <View style={styles.closeModalButtonContainer}>
          <TouchableRipple
            style={styles.rippleButtonModal}
            onPress={props.hideModal}
            rippleColor={props.chosenRippleColor}>
            <Text style={styles.rippleText}>Okay</Text>
          </TouchableRipple>
        </View>
      </Modal>
    </Portal>
  );
};

export default SettingsFeatures;

const styles = StyleSheet.create({
  closeModalButtonContainer: {
    marginBottom: 25,
    display: 'flex',
    alignContent: 'center',
    justifyContent: 'center',
    width: '100%',
  },
  rippleButtonModal: {
    padding: 4,
    borderRadius: 3,
    width: '100%',
    backgroundColor: themeColors.darkgray,
  },
  rippleText: {
    textAlign: 'center',
    letterSpacing: 0,
    color: 'white',
    fontWeight: 'bold',
  },
  featuresInfo: {
    alignContent: 'center',
    justifyContent: 'center',
    height: '100%',
    flex: 1,
    width: '100%',
  },
  featureText: {
    marginVertical: 5,
    textAlign: 'center',
    fontWeight: 'bold',
  },
});
