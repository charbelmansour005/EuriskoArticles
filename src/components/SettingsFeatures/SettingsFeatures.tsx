import {StyleSheet, Text, View, Modal} from 'react-native'
import {
  Portal,
  TouchableRipple,
  Provider as PProvider,
} from 'react-native-paper'
import {themeColors} from '../../helpers/themeColors'
import React from 'react'
import {SettingsFeaturesProps} from './types'

const SettingsFeatures = ({...props}: SettingsFeaturesProps): JSX.Element => {
  return (
    <PProvider>
      <Portal>
        <Modal visible={props?.isVisible} onDismiss={props?.hideModal}>
          <View style={styles.featuresInfo}>
            <Text
              style={{
                color: 'gray',
                ...styles.featureText,
              }}>
              Hold an article to visit website
            </Text>
            <Text
              style={{
                color: 'gray',
                ...styles.featureText,
              }}>
              Press on an article to see more details
            </Text>
            <Text
              style={{
                color: props?.chosenRippleColor,
                ...styles.featureText,
              }}>
              Dynamic colors
            </Text>
            <Text
              onPress={props?.testToast}
              style={{
                color: 'gray',
                ...styles.featureText,
              }}>
              A Welcome Toast (Press to test)
            </Text>
          </View>
          <View style={styles.closeModalButtonContainer}>
            <TouchableRipple
              style={styles.rippleButtonModal}
              onPress={props?.hideModal}
              rippleColor={props?.chosenRippleColor}>
              <Text style={styles.rippleText}>Okay</Text>
            </TouchableRipple>
          </View>
        </Modal>
      </Portal>
    </PProvider>
  )
}

export default SettingsFeatures

const styles = StyleSheet.create({
  closeModalButtonContainer: {
    marginBottom: 25,
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
    alignItems: 'center',
  },
  rippleButtonModal: {
    padding: 8,
    borderRadius: 3,
    width: '50%',
    backgroundColor: themeColors.darkgray,
  },
  rippleText: {
    textAlign: 'center',
    letterSpacing: 0,
    color: 'white',
    fontWeight: 'bold',
  },
  featuresInfo: {
    alignContent: 'center',
    justifyContent: 'center',
    height: '100%',
    flex: 1,
    width: '100%',
  },
  featureText: {
    marginVertical: 5,
    textAlign: 'center',
    fontWeight: 'bold',
  },
})
