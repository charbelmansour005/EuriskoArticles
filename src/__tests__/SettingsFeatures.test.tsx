import React from 'react'
import renderer from 'react-test-renderer'
import SettingsFeatures from '../components/SettingsFeatures/SettingsFeatures'
// passed
test('renders correctly', async () => {
  const tree = renderer.create(<SettingsFeatures />).toJSON()
  expect(tree).toMatchSnapshot()
})
