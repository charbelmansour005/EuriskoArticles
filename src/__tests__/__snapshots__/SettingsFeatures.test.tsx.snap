// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Modal
      hardwareAccelerated={false}
      visible={true}
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }
        >
          Hold an article to read more or visit website
        </Text>
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }
        >
          Press Read more on an article to open more details
        </Text>
        <Text
          style={
            Object {
              "color": undefined,
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }
        >
          Dynamic colors
        </Text>
        <Text
          style={
            Object {
              "color": "gray",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }
        >
          A Welcome Toast (Press to test)
        </Text>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
            "display": "flex",
            "justifyContent": "center",
            "marginBottom": 25,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              false,
              Object {
                "backgroundColor": "#565656",
                "borderRadius": 3,
                "padding": 4,
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "letterSpacing": 0,
                "textAlign": "center",
              }
            }
          >
            Okay
          </Text>
        </View>
      </View>
    </Modal>
  </View>
</RNCSafeAreaProvider>
`;
